### DISCUSSION TITLE
How to infer on an image without saving it to a file

### DISCUSSION THREAD

**Discussion by lidiahhhh** [open-edge-platform/anomalib#937] [Q&A] [answered]
Hello,
I would like to infer an image that is in RAM without saving it to a file and then creating a Dataset on the image path. 

Can I call predict directly on the image array? The image is the result of an object detection step.
Thanks

**Comment by tominator95** [comment, marked as answer]
Hi,
I created a REST API for inference. When I send the request, I send the file within it. So its not saved anywhere. The ```inferencer.predict()```  fucntion expects a path to an image file or the image as numpy array. More in the code snippet:

```python
from anomalib.deploy import TorchInferencer
from PIL import Image

inferencer = TorchInferencer(config="<<path to config>>", model_source="<<path to weights file>>", device="auto")

def predict(file: typing.IO[bytes]):
        file_bytes = bytearray(file.read())
        img = Image.open(io.BytesIO(file_bytes))
        img_arr = np.array(img)
       
       predictions = inferencer.predict(image=img_arr)
```
I hope that helps you.

**Reply by lidiahhhh** [reply]
Thanks! It really helped me
